usr/share/rear/finalize/GNU/Linux/260_rename_diskbyid.sh

```diff
@@ -57,37 +64,66 @@ while read ID DEV_NAME; do
     done
   fi
   echo $ID $DEV_NAME $ID_NEW
-done < $OLD_ID_FILE > $NEW_ID_FILE
+done < "$OLD_ID_FILE" > "$NEW_ID_FILE"
+
+sed_change_monitor="$TMP_DIR/by-id_change"
 
 for file in $FILES; do
-	realfile=$TARGET_FS_ROOT/$file
-	[ ! -f $realfile ] && continue	# if file is not there continue with next one
+	realfile="$TARGET_FS_ROOT/$file"
+	[ ! -f "$realfile" ] && continue	# if file is not there continue with next one
 	# keep backup
-	cp $realfile ${realfile}.rearbak
+	cp "$realfile" "${realfile}.rearbak"
 	# we should consider creating a sed script within a string
 	# and then call sed once (as done other times)
 	while read ID DEV_NAME ID_NEW; do
+		# initialize a sed_change variable to monitor if sed change a file
+		sed_change=0
+
 		if [ -n "$ID_NEW" ]; then
+
+      # If ID and ID_NEW are the same, no need to change, go the next device.
+      [[ "$ID" == "$ID_NEW" ]] && continue
+
 			# great, we found a new device
 			ID_FULL=/dev/disk/by-id/$ID
 			ID_NEW_FULL=/dev/disk/by-id/$ID_NEW
-			sed -i "s#$ID_FULL\([^-a-zA-Z0-9]\)#$ID_NEW_FULL\1#g" \
-				$realfile
+
+			# Using w flag to store changes made by sed in a output file $sed_change_monitor
+			# if no change is made, $sed_change_monitor will stay empty.
+			sed -i "s#$ID_FULL\([^-a-zA-Z0-9]\)#$ID_NEW_FULL\1#w $sed_change_monitor" \
+				"$realfile"
 			#                 ^^^^^^^^^^^^^^^
 			# This is to make sure we get the full ID (and not
 			# a substring) because we ask sed for a char other then
 			# those contained in IDs.
             # This does not work with IDs at line end: substitute also those:
-			sed -i "s#$ID_FULL\$#$ID_NEW_FULL#g" $realfile
+
+			# Check if sed made chage on the current file.
+			# (when size of $sed_change_monitor is not null)
+			test -s "$sed_change_monitor" && sed_change=1
+
+			sed -i "s#$ID_FULL\$#$ID_NEW_FULL#w $sed_change_monitor" $realfile
+			test -s "$sed_change_monitor" && sed_change=1
 		else
 			# lets try with the DEV_NAME as fallback
 			[ -z "$DEV_NAME" ] && continue
 			# not even DEV_NAME exists, we can't do anything
-			ID_FULL=/dev/disk/by-id/$ID
-			sed -i "s#$ID_FULL\([^-a-zA-Z0-9]\)#/dev/$DEV_NAME\1#g" \
-				$realfile
-			sed -i "s#$ID_FULL\$#/dev/$DEV_NAME#g" $realfile
+			ID_FULL=$ID
+			sed -i "s#$ID_FULL\([^-a-zA-Z0-9]\)#/dev/$DEV_NAME\1#w $sed_change_monitor" \
+				"$realfile"
+			test -s "$sed_change_monitor" && sed_change=1
+
+			sed -i "s#$ID_FULL\$#/dev/$DEV_NAME#w $sed_change_monitor" "$realfile"
+			test -s "$sed_change_monitor" && sed_change=1
```

Comment:
this file will be written by `sed` if changes are made into `$realfile`.
But it is overwritten (no append).
So if on the last `sed` didn't make any change, the file `$sed_change_monitor` will be empty even if a previous `sed` made a change.